sbi/neural_nets/embedding_nets.py

```diff
@@ -268,17 +273,32 @@ def forward(self, x: Tensor) -> Tensor:
         """
         batch, permutation_dim, _ = x.shape
 
-        iid_embeddings = self.trial_net(x.view(batch * permutation_dim, -1)).view(
-            batch, permutation_dim, -1
-        )
+        # if no NaNs for padding varying trial lengths we can batch the computation
+        if not torch.isnan(x).any():
+            trial_embeddings = self.trial_net(x.view(batch * permutation_dim, -1)).view(
+                batch, permutation_dim, -1
+            )
+            combined_embedding = self.combining_function(trial_embeddings, dim=1)
+            trial_counts = torch.ones(batch, 1, dtype=torch.float32) * permutation_dim
 
-        if self.combining_operation == "mean":
-            e = iid_embeddings.mean(dim=1)
-        elif self.combining_operation == "sum":
-            e = iid_embeddings.sum(dim=1)
+        # otherwise we need to loop over the batch to account for varying trial lengths
         else:
-            raise ValueError("combining_operation must be in ['sum', 'mean'].")
-
-        embedding = self.fc_subnet(e)
-
-        return embedding
+            combined_embedding = []
+            trial_counts = torch.zeros(batch, 1)
+            for i in range(batch):
```

Comment:
This will make things very slow, right?

Reply:
yes, scales with `training_batch_size` though, not with data size.